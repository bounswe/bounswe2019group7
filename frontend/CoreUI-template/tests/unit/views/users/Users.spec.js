import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Users from '@/views/users/Users'
import VueRouter from 'vue-router';

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(BootstrapVue)

describe('Users.vue', () => {
  it('has a name', () => {
    expect(Users.name).toMatch('Users')
  })
  it('has a created hook', () => {
    expect(typeof Users.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Users.data).toMatch('function')
    const defaultData = Users.data()
    expect(defaultData.currentPage).toBe(1)
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Users', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })
    expect(wrapper.is(Users)).toBe(true)
  })
  it('renders props.caption when passed', () => {
    const caption = 'Users List'
    const wrapper = mount(Users, {
      propsData: { caption },
      localVue,
      router
    })
    expect(wrapper.find('div.card-header > div').text()).toMatch(caption)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Users, {
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('should have methods', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })

    expect(typeof Users.methods.userLink  ).toEqual('function')
    expect(Users.methods.userLink(42)).toBe('users/42')
    expect(typeof Users.methods.rowClicked ).toEqual('function')
    expect(wrapper.vm.rowClicked({id:42})).toBeUndefined()
  })
})
