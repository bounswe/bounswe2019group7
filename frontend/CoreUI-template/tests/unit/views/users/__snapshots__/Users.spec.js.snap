// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users.vue renders correctly 1`] = `
<b-row-stub
  tag="div"
>
  <b-col-stub
    cols="12"
    tag="div"
    xl="6"
  >
    <b-card-stub
      bodytag="div"
      footertag="div"
      headertag="div"
      name="slide"
      subtitle=""
      subtitletag="h6"
      subtitletextvariant="muted"
      tag="div"
      title=""
      titletag="h4"
    >
      <div
        slot="header"
      >
        Users
      </div>
       
      <b-table-stub
        apiurl=""
        currentpage="1"
        emptyfilteredtext="There are no records matching your request"
        emptytext="There are no records to show"
        fields="[object Object],[object Object],[object Object],[object Object],[object Object]"
        hover="true"
        items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        labelsortasc="Click to sort Ascending"
        labelsortclear="Click to clear sorting"
        labelsortdesc="Click to sort Descending"
        perpage="5"
        responsive="sm"
        selectedvariant="primary"
        selectmode="multi"
        sortby=""
        sortdirection="asc"
        striped="true"
        value=""
      />
       
      <nav>
        <b-pagination-stub
          align="left"
          arialabel="Pagination"
          ellipsistext="…"
          firsttext="«"
          hidegotoendbuttons="true"
          labelfirstpage="Go to first page"
          labellastpage="Go to last page"
          labelnextpage="Go to next page"
          labelpage="Go to page"
          labelprevpage="Go to previous page"
          lasttext="»"
          limit="5"
          nexttext="Next"
          perpage="5"
          prevtext="Prev"
          size="sm"
          totalrows="26"
          value="1"
        />
      </nav>
    </b-card-stub>
  </b-col-stub>
</b-row-stub>
`;
